br/cmd/br/stream.go

```diff
@@ -122,19 +123,47 @@ func newStreamStatusComamnd() *cobra.Command {
 	return command
 }
 
+func newStreamRestoreCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "restore",
+		Short: "restore a stream backups",
+		Args:  cobra.NoArgs,
+		RunE: func(command *cobra.Command, _ []string) error {
+			return streamCommand(command, task.StreamRestore)
+		},
+	}
+	task.DefineFilterFlags(command, acceptAllTables)
+	task.DefineStreamRestoreFlags(command.Flags())
+	return command
+}
+
 func streamCommand(command *cobra.Command, cmdName string) error {
 	var cfg task.StreamConfig
-	if err := cfg.ParseCommonFromFlags(command.Flags()); err != nil {
-		command.SilenceUsage = false
+	var err error
+	defer func() {
+		if err != nil {
+			command.SilenceUsage = false
+		}
+	}()
+	if err = cfg.Config.ParseFromFlags(command.Flags()); err != nil {
 		return errors.Trace(err)
 	}
 
-	if cmdName == task.StreamStart {
-		if err := cfg.ParseStreamStartFromFlags(command.Flags()); err != nil {
+	switch cmdName {
+	case task.StreamRestore:
+		if err = cfg.ParseStreamRestoreFromFlags(command.Flags()); err != nil {
+			return errors.Trace(err)
+		}
+	case task.StreamStart:
+		if err = cfg.ParseStreamStartFromFlags(command.Flags()); err != nil {
+			return errors.Trace(err)
+		}
+		fallthrough
+	default:
+		if err = cfg.ParseStreamCommonFromFlags(command.Flags()); err != nil {
 			return errors.Trace(err)
 		}
```

Comment:
Maybe remove the `fallthrough` and extract the `PraseStreamCommonFromFlags` out to the `switch`?

Reply:
`PraseStreamCommonFromFlags` parse `task-name` config. which is not used is stream restore.🤔. but used for stream start. it cannot be simply move outside.

Reply:
Already add a TODO.